src/acceleration/BaseQNAcceleration.hpp

```diff
@@ -104,7 +106,16 @@ class BaseQNAcceleration : public Acceleration {
    * Has to be called after every implicit coupling iteration.
    */
   virtual void performAcceleration(DataMap &cplData);
-
+  /**
+   * @brief Transform the bounded input data into the inf range.
+   */
+  virtual void projectFWInput(DataMap &cplData, const std::vector<DataID> &dataIDs, std::map<int, double> lowerBounds,
+                              std::map<int, double> upperBounds);
+  /**
+   * @brief Trandform the output data backward from inf to certain range.
+   */
+  virtual void projectBWInput(DataMap &cplData, const std::vector<DataID> &dataIDs, std::map<int, double> lowerBounds,
```

Comment:
```suggestion
  virtual void projectBWOutput(DataMap &cplData, const std::vector<DataID> &dataIDs, std::map<int, double> lowerBounds,
```
?

And let's not use abbreviations like "BW" or "FW". 
"transform" instead of "project"?